tc/sgx/trusted_worker_manager/enclave/signup_enclave_wpe.cpp

```diff
@@ -44,6 +44,29 @@ static void CreateSignupReportDataWPE(const uint8_t* ext_data,
     EnclaveData* enclave_data,
     sgx_report_data_t* report_data);
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
+    tcf_err_t result = TCF_SUCCESS;
+    tcf::error::ThrowIf<tcf::error::ValueError>(in_nonce_size <= 0,
+        "Nonce size should be positive value");
+    try {
+        out_nonce = tcf::crypto::RandomBitString(in_nonce_size).data();
```

Comment:
WPE enclave has to store the nonce inside WPE enclave and verify the unique ID signature from KME using this nonce.

Reply:
Done.